### PR TITLE
Fix untested MismatchingDataValueTypeException

### PR DESCRIPTION
**Pull request by thiemowmde**
It seems this test was never executed. Note the actual bug in the class (the single-quoted `$message`).

### CODE REVIEW

File: src/ValueFormatters/Exceptions/MismatchingDataValueTypeException.php [hunk 1]
```diff
@@ -29,14 +30,19 @@ class MismatchingDataValueTypeException extends FormattingException {
...
+		$dataValueType,
+		$message = '',
 		Exception $previous = null
 	) {
 		$this->expectedValueType = $expectedValueType;
 		$this->dataValueType = $dataValueType;
 
-		$message = '$dataValueType "' . $dataValueType . '" does not match $expectedValueType "'
-			. $expectedValueType . '": $message';
+		if ( $message === '' ) {
```
Comment (mariushoch) [inline src/ValueFormatters/Exceptions/MismatchingDataValueTypeException.php]: Why did you change this to no longer prefix the message passed as argument?

Comment (thiemowmde) [inline src/ValueFormatters/Exceptions/MismatchingDataValueTypeException.php]: Because this feature is unused anyway and also provides no benefit. Other exceptions do have similar code. They all (as far as I'm aware of) do the same: Use the custom message, if provided. Construct a message otherwise. But do not combine the two.

Comment (filbertkm) [inline src/ValueFormatters/Exceptions/MismatchingDataValueTypeException.php]: change is ok with me
